storage/templates.go

```diff
@@ -27,3 +27,22 @@ func (db *store) GetTemplate(id, userID int64) (*entities.Template, error) {
 	err := db.Where("user_id = ? and id = ?", userID, id).Find(template).Error
 	return template, err
 }
+
+// ListTemplates fetches templates by user id, and populates the pagination obj
+func (db *store) ListTemplates(userID int64, p *PaginationCursor, scopeMap map[string]string) error {
+	p.SetCollection(&[]entities.TemplatesCollection{})
+	p.SetResource("templates")
+
+	for k, v := range scopeMap {
+		if k == "name" {
```

Comment:
@gudgl @FilipNikolovski  do we need filter here?

Reply:
Sure.